components/AIEstimation.tsx

```diff
@@ -0,0 +1,60 @@
+// components/ai-estimation.tsx
+'use client'
+
+import { useState } from 'react'
+import { useMutation } from 'convex/react'
+import { api } from '@/convex/_generated/api'
+import { Button } from "@/components/ui/button"
+import { Card, CardContent, CardHeader, CardTitle } from "@/components/ui/card"
+import { Spinner } from "@/components/ui/spinner"
+import { useToast } from '@/hooks/use-toast'
+
+interface AIEstimationProps {
+    vehicleDetails: any // Replace with proper type
+    selectedServices: string[]

```

Comment:
**Suggestion:** Use a more descriptive type than `string[]` for `selectedServices` to ensure that the services are valid and to provide better type checking throughout the application. [best practice, importance: 8]
```suggestion
    selectedServices: Array<ServiceType>
```